scripts/mod_loader/modapi/vector.lua

```diff
@@ -0,0 +1,306 @@
+
+math.vec2 = {}
+math.vec3 = {}
+math.vec4 = {}
+
+local mt_vec2 = {
+	x = 1, y = 2,
+	[1] = 1, [2] = 2
+}
+
+local mt_vec3 = {
+	x = 1, y = 2, z = 3,
+	i = 1, j = 2, k = 3,
+	[1] = 1, [2] = 2, [3] = 3
+}
+
+local mt_vec4 = {
+	top = 1, bottom = 2, left = 3, right = 4,
+	north = 1, south = 2, west = 3, east = 4,
+	n = 1, s = 2, w = 3, e = 4,
+	t = 1, b = 2, l = 3, r = 4,
+	x = 1, y = 2, z = 3, time = 4,
+	i = 1, j = 2, k = 3,
+	[1] = 1, [2] = 2, [3] = 3, [4] = 4
+}
+
+local function addVectors(a, b)
+	
+	if #a == 4 or #b == 4 then
+		return math.vec4(
+			a[1] + b[1],
+			a[2] + b[2],
+			a[3] + b[3],
+			a[4] + b[4]
+		)
+		
+	elseif #a == 3 or #b == 3 then
+		return math.vec3(
+			a[1] + b[1],
+			a[2] + b[2],
+			a[3] + b[3]
+		)
+		
+	elseif #a == 2 or #b == 2 then
+		return math.vec2(
+			a[1] + b[1],
+			a[2] + b[2]
+		)
+	end
+	
+	return nil
+end
+
+local function subVectors(a, b)
+	return addVectors(a, b(-b[1], -b[2], -b[3], -b[4]))
```

Comment:
I feel using `-b` would be better in this case, overriding `__unm` in the metatable so `-b` negates the vector.

Reply:
I totally agree - I Didn't know __unm existed.

Reply:
Added Unary Minus